src/controller_examples/rabbitmq_controller/proof/liveness/liveness.rs

```diff
@@ -50,12 +67,23 @@ proof fn liveness_proof_forall_rabbitmq()
     };
 }
 
+spec fn assumption_and_invariants_of_all_phases(rabbitmq: RabbitmqClusterView) -> TempPred<RMQCluster> {
+    invariants(rabbitmq)
+    .and(always(lift_state(RMQCluster::desired_state_is(rabbitmq))))
+    .and(invariants_since_phase_I(rabbitmq))
+    .and(invariants_since_phase_II(rabbitmq))
+    .and(invariants_since_phase_III(rabbitmq))
+    .and(invariants_since_phase_IV(rabbitmq))
+    .and(invariants_since_phase_V(rabbitmq))
+}
+
 // Next and all the wf conditions.
 spec fn next_with_wf() -> TempPred<RMQCluster> {
```

Comment:
We probably should just move it to the cluster state machine and replace that `partial_spec` (if that still exists).